javascript/utilities/src/utils/files.ts

```diff
@@ -178,10 +178,13 @@ export function renameDuplicatedFileName(
 	map: Map<string, number>,
 	name: string,
 ): string {
-	const next = (map.get(name) || 0) + 1
-	map.set(name, next)
-	const ext = extension(name)
-	return next > 1 ? `${name.replace(`.${ext}`, '')} (${next}).${ext}` : name
+	const cleanName = cleanFileName(name)
+	const next = (map.get(cleanName) || 0) + 1
+	map.set(cleanName, next)
+	const ext = extension(cleanName)
+	return next > 1
+		? `${cleanName.replace(`.${ext}`, '')} (${next}).${ext}`
```

Comment:
let's introduce a variable to make this clearer; 

`const basename = cleanName.replace(`.${ext}`, '')`